libpromises/files_names.c

```diff
@@ -568,8 +572,7 @@ int CompressPath(char *dest, const char *src)
 
         AddSlash(dest);
 
-        /* TODO use dest_size parameter instead of CF_BUFSIZE. */
-        size_t ret = strlcat(dest, node, CF_BUFSIZE);
+        size_t ret = strlcat(dest, node, dest_length);
 
         if (ret >= CF_BUFSIZE)
         {
```

Comment:
I can see that `CompressPath()` returned value is not checked in most places. The least we can do is `dest[0]='\0'` so that we avoid deleting the parent directory of an extremely long path.